aiohttp/connector.py

```diff
@@ -964,6 +971,18 @@ async def _wrap_create_connection(
             async with ceil_timeout(
                 timeout.sock_connect, ceil_threshold=timeout.ceil_threshold
             ):
+                if self._network_interface:
+                    kwargs["sock"] = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
It's best to prepare the socket through a dedicated variable and only then assign it. Moreover, I'd move the init+connect to a helper function and just assign the result here.